pandas/tests/groupby/aggregate/test_aggregate.py

```diff
@@ -1603,3 +1603,20 @@ def test_agg_with_as_index_false_with_list():
         columns=MultiIndex.from_tuples([("a1", ""), ("a2", ""), ("b", "sum")]),
     )
     tm.assert_frame_equal(result, expected)
+
+
+def test_agg_arrow_type():
+    df = DataFrame.from_dict(
+        {
+            "category": ["A"] * 10 + ["B"] * 10,
+            "bool_numpy": [True] * 5 + [False] * 5 + [True] * 5 + [False] * 5,
+        }
+    )
+    df["bool_arrow"] = df["bool_numpy"].astype("bool[pyarrow]")
+    result = df.groupby("category").agg(lambda x: x.sum() / x.count())
+    expected = DataFrame(
+        [[0.5, 0.5], [0.5, 0.5]],
+        columns=["bool_numpy", "bool_arrow"],
+        index=Index(["A", "B"], name="category"),
+    )
```

Comment:
Shouldn't the bool_arrow result here by `double[pyarrow]`?

Reply:
No, it is float64. The result DataFrame's bool_arrow is also float64. So in age_series function, if preserve_type is False, the pyarrow result would be same as numpy result.

Reply:
I'm asking what it should be. If I add a collection of `bool[pyarrow]`, I expect to get `int64[pyarrow]`. Likewise, if I divide two `int64[pyarrow]`, I expect to get `double[pyarrow]`.

Reply:
Oh, yeah. It should be double[pyarrow]. So we need to cast the numpy result to pyarrow types. I am thinking the current cast function "maybe_cast_pointwise_result"  need to cast numpy result array to corresponding pyarrow array, not original dtype.